charmcraft/main.py

```diff
@@ -95,11 +97,14 @@ def run(self, parsed_args, all_commands):
 ]
 
 
-# global options: the flag used internally, short and long parameters, and the help text
-GLOBAL_FLAGS = [
-    ('help', '-h', '--help', "Show this help message and exit"),
-    ('verbose', '-v', '--verbose', "Show debug information and be more verbose"),
-    ('quiet', '-q', '--quiet', "Only show warnings and errors, not progress"),
+# global options: the name used internally, its type, short and long parameters, and help text
+_Global = namedtuple('Global', 'name type short_option long_option help_message')
```

Comment:
Not important, but dataclass (or attr) would be a better choice than namedtuple imo (type annotations, etc.)

Reply:
It's simpler like this and good enough for now, we can improve it in the future.